z3/src/tactic.rs

```diff
@@ -0,0 +1,159 @@
+use std::ffi::{CStr, CString};
+use std::fmt;
+
+use z3_sys::*;
+use Context;
+use Goal;
+use Params;
+use Tactic;
+use Z3_MUTEX;
+
+impl<'ctx> Tactic<'ctx> {
+    pub fn list_all(ctx: &'ctx Context) {
+        let p = unsafe {
+            Z3_get_num_tactics(ctx.z3_ctx)
+        };
+        for n in 0..p {
+            let t = unsafe {
+                Z3_get_tactic_name(ctx.z3_ctx, n)
+            };
+            match unsafe { CStr::from_ptr(t) }.to_str() {
+                Ok(s) => println!("{}", s),
+                Err(_) => (),
+            }
+        }
+    }
+
+    fn new_from_z3(ctx: &'ctx Context, z3_tactic: Z3_tactic) -> Tactic<'ctx> {
+        Tactic {
+            ctx,
+            z3_tactic,
+        }
+    }
+
+    pub fn new(ctx: &'ctx Context, name: &'ctx str) -> Tactic<'ctx> {
```

Comment:
Can `&'ctx str` be changed to `&str`? Since the str lifetime may not be as short/long as the ctx lifetime, and we're not keeping that reference in the returned tactic struct